File: lista de exercicios/questão 15/questao.java
class SaldoInsuficienteException extends Exception {
    public SaldoInsuficienteException(double saldo, double valor) {
        super(String.format("Saldo insuficiente. Saldo atual: R$%.2f, Tentativa de saque: R$%.2f", saldo, valor));
    }
}

class ContaBancaria {
    private String titular;
    private double saldo;

    public ContaBancaria(String titular, double saldo) {
        this.titular = titular;
        this.saldo = saldo;
    }

    public void sacar(double valor) throws SaldoInsuficienteException {
        if (valor > saldo) {
            throw new SaldoInsuficienteException(saldo, valor);
        }
        saldo -= valor;
        System.out.printf("Saque de R$%.2f realizado com sucesso.%n", valor);
    }
    
    public static void main(String[] args) {
        ContaBancaria conta = new ContaBancaria("João", 100);

        try {
            conta.sacar(200);  // Exceção será lançada aqui
        } catch (SaldoInsuficienteException e) {
            System.out.println(e.getMessage());
        }
    }
}

File: lista de exercicios/questão 15/questao.py
class SaldoInsuficienteException(Exception):
    def __init__(self, saldo, valor):
        super().__init__(f"Saldo insuficiente. Saldo atual: R${saldo}, Tentativa de saque: R${valor}")

class ContaBancaria:
    def __init__(self, titular, saldo=0):
        self.titular = titular
        self.saldo = saldo

    def sacar(self, valor):
        if valor > self.saldo:
            raise SaldoInsuficienteException(self.saldo, valor)
        self.saldo -= valor
        print(f"Saque de R${valor} realizado com sucesso.")

# Exemplo de uso
conta = ContaBancaria("João", 100)

try:
    conta.sacar(200)  # Exceção será lançada aqui
except SaldoInsuficienteException as e:
    print(e)
